gitpod-io/gitpod
# [OIDC] Enable mark client config as "active"

## Description

This adds a "active" field to OIDC config records to be used to answer the following questions:
 * is a given SSO configuration ready for sign-ins by members of the owning Org
 * is a Dedicated Cell in "onboarding mode", i.e. is SSO not yet set up.

Once the OIDC flow was completed with the `activate` flag set, we should be able to check if the config is active. For instance, this is result the clients would get:

<img width="282" alt="Screenshot 2023-04-26 at 11 57 09" src="https://user-images.githubusercontent.com/914497/234541081-64422459-d52b-4f53-8295-cd434a8ce985.png">


## Related Issue(s)

Part of WEB-166

## How to test
* Setup Org with SSO in a preview env
* Get the regular sign-in URL from `[...]` menu in the configs list
* Append `activate=true` and check the state in the DB afterwards

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - at-oidc-test</li>
	<li><b>🔗 URL</b> - <a href="https://at-oidc-test.preview.gitpod-dev.com/workspaces" target="_blank">at-oidc-test.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

</details>

<details>
<summary>Installer Options</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold
